retail/interactive-tutorial/RetailProducts.Samples/SetInventorySample.cs

```diff
@@ -0,0 +1,162 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_set_inventory]
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using Google.Protobuf.WellKnownTypes;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Threading;
+
+/// <summary>
+/// The set inventory sample class.
+/// </summary>
+public class SetInventorySample
+{
+    /// <summary>
+    /// Get the retail product with inventory info.
+    /// </summary>
+    /// <param name="productName">The actual name of the retail product.</param>
+    /// <returns>Retail product with updated inventory info.</returns>
+    private static Product GetProductWithInventoryInfo(string productName)
+    {
+        var priceInfo = new PriceInfo
+        {
+            Price = 15.0f,
+            OriginalPrice = 20.0f,
+            Cost = 8.0f,
+            CurrencyCode = "USD"
+        };
+
+        var fulfillmentInfo = new FulfillmentInfo
+        {
+            Type = "pickup-in-store"
+        };
+
+        string[] placeIds = { "store1", "store2" };
+
+        fulfillmentInfo.PlaceIds.Add(placeIds);
+
+        var product = new Product
+        {
+            Name = productName,
+            PriceInfo = priceInfo,
+            Availability = Product.Types.Availability.InStock
+        };
+
+        product.FulfillmentInfo.Add(fulfillmentInfo);
+
+        return product;
+    }
+
+    /// <summary>
+    /// Get the set inventory request.
+    /// </summary>
+    /// <param name="productName">The actual product with inventory info.</param>
+    /// <returns>Set inventory request.</returns>
+    private static SetInventoryRequest GetSetInventoryRequest(string productName)
+    {
+        // The request timestamp
+        DateTime requestTimeStamp = DateTime.Now.ToUniversalTime();
+
+        // The out-of-order request timestamp
+        // request_time = datetime.datetime.now() - datetime.timedelta(days=1)
+
+        string[] paths = { "price_info", "availability", "fulfillment_info", "available_quantity" };
+        var setMask = new FieldMask();
+        setMask.Paths.AddRange(paths);
+
+        var setInventoryRequest = new SetInventoryRequest
+        {
+            Inventory = GetProductWithInventoryInfo(productName),
+            SetTime = Timestamp.FromDateTime(requestTimeStamp),
+            AllowMissing = true,
+            SetMask = setMask
+        };
+
+        var jsonSerializeSettings = new JsonSerializerSettings
+        {
+            NullValueHandling = NullValueHandling.Ignore,
+            ContractResolver = new CamelCasePropertyNamesContractResolver(),
+            Formatting = Formatting.Indented
+        };
+
+        var setInventoryRequestJson = JsonConvert.SerializeObject(setInventoryRequest, jsonSerializeSettings);
+
+        Console.WriteLine("\nSet inventory request: \n\n" + setInventoryRequestJson);
```

Comment:
Please don't convert the request to JSON for the purposes of writing it to the console, users won't need to do this, look at the rest of the Retail samples and just do `Console.WriteLine` for the relevant fields you want to print.

Reply:
done